### PR TITLE
[bug-fix] Fix roles retrieve issue in Group creation flow

### PR DESCRIPTION
**Pull request by JayaShakthi97**
### Purpose

$subject

### Related Issues

- https://github.com/wso2/product-is/issues/21422

### Related PRs

- N/A

### Checklist

- [ ] e2e cypress tests locally verified. (for internal contributers)
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Relevant backend changes deployed and verified
- [ ] [Unit tests](/docs/testing/UNIT_TESTING.md) provided. (Add links if there are any)
- [ ] [Integration tests](https://github.com/wso2/product-is/tree/master/modules/integration) provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines
- [ ] Ran FindSecurityBugs plugin and verified report
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets

### CODE REVIEW

File: features/admin.groups.v1/components/wizard/create-group-wizard.tsx [hunk 1]
```diff
@@ -167,34 +192,20 @@ export const CreateGroupWizard: FunctionComponent<CreateGroupProps> =
 
             setSubmitStep(WizardStepsFormTypes.BASIC_DETAILS);
         }
+    }, []);
 
-        if (roleList?.length < 1) {
-            if (isFirstLevelOrganization()) {
```
Comment (pavinduLakshan) [inline features/admin.groups.v1/components/wizard/create-group-wizard.tsx:172]: What is the reason for removing this logic?

File: features/admin.groups.v1/components/wizard/create-group-wizard.tsx [hunk 2]
```diff
@@ -152,6 +143,40 @@ export const CreateGroupWizard: FunctionComponent<CreateGroupProps> =
...
+
+    const consoleId: string = filteredApplicationList?.applications[0]?.id;
+
+    const {
+        data: fetchedRoleList,
+        error: rolesListError
+    } = useGetRolesList(
+        100,
+        null,
+        `audience.value ne ${consoleId}`,
```
Comment (pavinduLakshan) [inline features/admin.groups.v1/components/wizard/create-group-wizard.tsx:159]: Is it intentional to update this component's behaviour to work only for console application?

File: features/admin.groups.v1/components/wizard/create-group-wizard.tsx (hunk 1 above)
Comment (JayaShakthi97) [reply features/admin.groups.v1/components/wizard/create-group-wizard.tsx:172]: This is no longer needed. Moving forward `scim2` endpoint should be used. `organizationRoles` endpoint is not used. Mentioned in the issue description.

File: features/admin.groups.v1/components/wizard/create-group-wizard.tsx (hunk 2 above)
Comment (JayaShakthi97) [reply features/admin.groups.v1/components/wizard/create-group-wizard.tsx:159]: The intention is to filter out console app roles. Please check the existing behavior of the roles listing.

**Review by pavinduLakshan — DISMISSED** [review]

**Review by pavinduLakshan — APPROVED** [review]

**Pull request merged**